### ISSUE TITLE
should_not_remember_docs_in_index_after_backup_restore...*failed*

### ISSUE THREAD

**Issue by wohali** [apache/couchdb#565]
## Expected and Current Behaviour
Usually the `couchdb_views_tests` test `should_not_remember_docs_in_index_after_backup_restore` passes. Sometimes it fails.

Environment: Travis, Erlang 18.3

Makefile output:
```
  Upgrade and bugs related tests
    couchdb_views_tests:159: should_not_remember_docs_in_index_after_backup_restore...*failed*
in function couchdb_views_tests:'-query_view/4-fun-0-'/2 (test/couchdb_views_tests.erl, line 514)
in call from couchdb_views_tests:query_view/4 (test/couchdb_views_tests.erl, line 514)
in call from couchdb_views_tests:'-should_not_remember_docs_in_index_after_backup_restore/1-fun-8-'/1 (test/couchdb_views_tests.erl, line 173)
**error:{assertEqual,[{module,couchdb_views_tests},
              {line,514},
              {expression,"Code"},
              {expected,200},
              {value,500}]}
  output:<<"">>
```

couch.log output:
```
[info] 2017-06-02T23:12:29.470701Z nonode@nohost <0.29000.0> -------- alarm_handler: {set,{system_memory_high_watermark,[]}}
[info] 2017-06-02T23:12:29.508603Z nonode@nohost <0.29086.0> -------- Opening index for db: eunit-test-db-1496445149424073 idx: _design/foo sig: "5398c9ad12cad56d2228cc89a2770ddf"
[info] 2017-06-02T23:12:29.508945Z nonode@nohost <0.29089.0> -------- Starting index update for db: eunit-test-db-1496445149424073 idx: _design/foo
[info] 2017-06-02T23:12:29.517038Z nonode@nohost <0.29089.0> -------- Index update finished for db: eunit-test-db-1496445149424073 idx: _design/foo
[notice] 2017-06-02T23:12:29.517765Z nonode@nohost <0.29036.0> -------- 127.0.0.1 - - GET /eunit-test-db-1496445149424073/_design/foo/_view/bar 200
[info] 2017-06-02T23:12:29.518975Z nonode@nohost <0.29086.0> -------- Index shutdown by monitor notice for db: eunit-test-db-1496445149424073 idx: _design/foo
[info] 2017-06-02T23:12:29.519120Z nonode@nohost <0.29022.0> -------- db eunit-test-db-1496445149424073 died with reason shutdown
[info] 2017-06-02T23:12:29.519432Z nonode@nohost <0.29086.0> -------- Closing index for db: eunit-test-db-1496445149424073 idx: _design/foo sig: "5398c9ad12cad56d2228cc89a2770ddf" because normal
[error] 2017-06-02T23:12:29.520706Z nonode@nohost <0.29037.0> -------- Uncaught error in HTTP request: {error,{badmatch,{'EXIT',noproc}}}
[info] 2017-06-02T23:12:29.522215Z nonode@nohost <0.29037.0> -------- Stacktrace: [{couch_file,pread_binary,2,[{file,"src/couch_file.erl"},{line,168}]},{couch_file,pread_term,2,[{file,"src/couch_file.erl"},{line,156}]},{couch_btree,get_node,2,[{file,"src/couch_btree.erl"},{line,423}]},{couch_btree,lookup,3,[{file,"src/couch_btree.erl"},{line,282}]},{couch_btree,lookup,2,[{file,"src/couch_btree.erl"},{line,272}]},{couch_db,get_full_doc_info,2,[{file,"src/couch_db.erl"},{line,286}]},{couch_db,open_doc_int,3,[{file,"src/couch_db.erl"},{line,1366}]},{couch_db,open_doc,3,[{file,"src/couch_db.erl"},{line,189}]}]
[notice] 2017-06-02T23:12:29.522460Z nonode@nohost <0.29037.0> -------- 127.0.0.1 - - GET /eunit-test-db-1496445149424073/_design/foo/_view/bar 500
[error] 2017-06-02T23:12:29.522708Z nonode@nohost <0.29037.0> -------- httpd 500 error response:
 {"error":"badmatch","reason":"{'EXIT',noproc}"}
[info] 2017-06-02T23:12:29.532104Z nonode@nohost <0.7.0> -------- Application couch exited with reason: stopped
[info] 2017-06-02T23:12:29.533589Z nonode@nohost <0.29000.0> -------- alarm_handler: {clear,system_memory_high_watermark}
```

## Possible Solution
Race condition in test? @davisp ?

**Comment by nickva** [comment]
Have we seen these after this commit https://github.com/apache/couchdb/commit/05d2e1e6a7db3e2526530cdbb446034557b94e57 ?

Wondering if that fixed this as well

**Comment by eiri** [comment]
Ok the problem here is not in a view, but in couch_server. During the test we are shutting down old db updater, swapping db file and then doing view query. In reality couch_server could be a bit slow on evicting old updater pid from its cache (as I found 5 ms is quite enough), so the query is getting old pid and naturally crashes with `noproc`.

I've opened PR #600 to make sure we have new updater started.
